aten/src/ATen/core/jit_type.h

```diff
@@ -487,6 +487,16 @@ struct CAFFE2_API ListType : public SingleElementType<TypeKind::ListType, ListTy
   static ListTypePtr create( T&& ... all ) {
     return ListTypePtr(new ListType( std::forward<T>(all)... )); // NOLINT(modernize-make-shared)
   }
+  bool isSubtypeOf(const TypePtr rhs) const override {
```

Comment:
This does not seem right. There should be no special rules for lists and optionals. This diff is suppose to reduce special cases, not add them.